react/components/StructuredData.js

```diff
@@ -0,0 +1,151 @@
+    console.warn('Failed to parse multilanguage product description')
+  }
+  return parsedDescription || description
+}
+
+const lowestPriceItem = compose(
+  head,
+  sort((itemA, itemB) => (path(['seller', 'commertialOffer', 'Price'], itemA) - path(['seller', 'commertialOffer', 'Price'], itemB)))
+)
+
+const highestPriceItem = compose(
+  last,
+  sort((itemA, itemB) => (path(['seller', 'commertialOffer', 'Price'], itemA) - path(['seller', 'commertialOffer', 'Price'], itemB)))
+)
+
+const priceItems = items => {
+  const lowPrice = lowestPriceItem(items)
+  const highPrice = highestPriceItem(items)
+  return {
+    lowPrice,
+    highPrice,
+  }
+}
+
+const parseSKUToOffer = (item, currency) => {
+
+  const { seller } = lowestPriceInStockSKU(item)
+  const calculateAvailability = seller => {
+    return (
+      pathOr(ITEM_AVAILABLE, ['commertialOfferlowestPriceInStock', 'AvailableQuantity'], seller)
+        ? "http://schema.org/InStock" : "http://schema.org/OutOfStock")
+  }
+
+  const offer = `{
+    "@type": "Offer", 
+    "price": "${path(['commertialOffer', 'Price'], seller)}",
+    "priceCurrency": "${currency}",
+    "availability": "${calculateAvailability(seller)}",
+    "sku": "${item.itemId}",
+    "itemCondition": "http://schema.org/NewCondition",
+    "priceValidUntil": "${path(['commertialOffer', 'PriceValidUntil'], seller)}",
+    "seller": {
+      "@type": "Organization", 
+      "name": "${seller.sellerName}"
+    }
+  }`
+
+  return offer
+}
+
+const composeAggregateOffer = (product, currency) => {
+  const items = map(item => ({
+    item,
+    seller: lowestPriceInStockSeller(item)
+  }), product.items)
+
+  const { lowPrice, highPrice } = priceItems(items)
+  const offersList = map((element, index) => {
+    return parseSKUToOffer(element, currency)
+  }, product.items)
+
+  const aggregateOffer = `{
+    "@type": "AggregateOffer", 
+    "lowPrice": "${path(['seller', 'commertialOffer', 'Price'], lowPrice)}",
+    "highPrice": "${path(['seller', 'commertialOffer', 'Price'], highPrice)}",
+    "priceCurrency": "${currency}",
+    "offers": [${offersList}],
+    "offerCount": "${length(items)}"
+  }`
+
+  return aggregateOffer
+}
+
+const parseToJsonLD = (product, query, currency, locale) => {
+  const skuId = query.skuId || path(['items', '0', 'itemId'], product)
+  const image = head(path(['items', '0', 'images'], product))
+  const brand = product.brand
+  const name = product.productName
+  const description = tryParsingLocale(product.description, locale)
+
+  const productLD = `{
+    "@context": "https://schema.org/",
+    "@type" : "Product",
+    "name": "${name}", 
+    "brand": "${brand}",
+    "image": "${image.imageUrl}",
+    "description": "${description}",
+    "mpn": "${product.productId}",
+    "sku": "${skuId}",
+    "offers": ${composeAggregateOffer(product, currency)}
+    
+  }`
+
+  return productLD
+}
+
+export default function StructuredData({ product, query }, { culture: { currency, locale } }) {
```

Comment:
Export the component at the bottom of the page, it improves readability

Reply:
And also, get that properties that you are getting with the [legacy API](https://reactjs.org/docs/legacy-context.html) with `useRuntime`. Refer [here](https://github.com/vtex-apps/store-header/blob/5b995ce600cf4ed06c2e4d837f196750ddf95c08/react/hooks/useDevice.js#L10) for usage